lib/workers/repository/updates/branch-name.ts

```diff
@@ -0,0 +1,54 @@
+import { clean as cleanGitRef } from 'clean-git-ref';
+import slugify from 'slugify';
+import { RenovateConfig } from '../../../config/common';
+import { logger } from '../../../logger';
+import { clone } from '../../../util/clone';
+import * as template from '../../../util/template';
+
+/**
+ * Clean git branch name
+ *
+ * Remove what clean-git-ref fails to:
+ * - leading dot/leading dot after slash
+ * - trailing dot
+ * - whitespace
+ */
+function cleanBranchName(branchName: string): string {
+  return cleanGitRef(branchName)
+    .replace(/^\.|\.$/, '') // leading or trailing dot
+    .replace(/\/\./g, '/') // leading dot after slash
+    .replace(/\s/g, ''); // whitespace
+}
+
+export function getBranchName(update_: RenovateConfig): string {
+  const update = clone(update_);
```

Comment:
This clone would make all changes on `update` obsolete. Do we really want to clone here? We would loos all new update props.

If we don't need these old props (`groupSlug`, ...), remove them.

Reply:
We don't need groupSlug after this function is done. I was trying to avoid reassigning parameters.

Reply:
so can we simply remove all reassignments? (use local vars if needed)

Reply:
Too complex considering that `upgrade` needs to be passed to compile functions multiple times. Refactored the code anyway.

Reply:
ok, let me know when to review again.